src/components/ProductsPagination/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import './styles.scss'
+import { FeaturedData } from '../../interfaces/Featured'
+
+interface ProductsPaginationProps {
+  products: FeaturedData;
+  onHandleProductFetch: Function
+}
+
+interface ButtonProp {
+  label: number
+}
+
+export default function ProductsPagination ({ products, onHandleProductFetch }: ProductsPaginationProps) {
+  const [paginationButtons, setPaginationButtons] = useState<Array<ButtonProp>>([])
+
+  useEffect(() => {
+    const newButtons = Array
+      .from({ length: (products.total_pages as number) }, (_, i) => i + 1)
+      .map(label => ({ label }))
+    setPaginationButtons(newButtons)
+  }, [products])
+
+  const getPage = (link: string) => {
+    if (!link) {
+      return null
+    }
+    const url = new URL(link)
+    return url.searchParams.get('page')
+  }
+
+  return (
+    <div
+      className='products-pagination'
+    >
+      <button
+        disabled={!products.prev_page}
+        onClick={() => onHandleProductFetch(getPage((products.prev_page as string)))}
```

Comment:
When using TS, be aware of using type assertion 'as', is a best practice to use it as a last approach, is better to type the getPage function